dev-server/serve-dev.js

```diff
@@ -21,28 +21,41 @@ const TEMPLATE_PATH = `${__dirname}/templates/`;
  */
 
 /**
- * Generate `<script>` content for client configuration and injection
+ * Render client config and script embed
  *
- * @param {string} clientUrl
- * @return {string}
+ * @param {Object} context
  */
-function renderConfig(clientUrl) {
-  const scriptTemplate = fs.readFileSync(
-    `${TEMPLATE_PATH}client-config.js.mustache`,
-    'utf-8'
-  );
-  return Mustache.render(scriptTemplate, { clientUrl });
+function renderScript(context) {
+  const scriptTemplate = `
+    {{{hypothesisConfig}}}
+    <script>
+    const embedScript = document.createElement('script');
+    embedScript.src = '{{{clientUrl}}}'.replace('{current_host}', document.location.hostname);
+    document.body.appendChild(embedScript);
+    </script>
+  `;
+  return Mustache.render(scriptTemplate, context);
 }
 
 /**
- * Build context for rendering templates in the defined views directory. This
- * is dead simple at present but could be extended as needs grow.
+ * Build context for rendering templates in the defined views directory.
  *
  * @param {Config} config
  */
 function templateContext(config) {
+  // Just the config by itself, in contrast with `hypothesisScript`, which
+  // combines this config with a <script> that adds the embed script
+  const hypothesisConfig = fs.readFileSync(
+    `${TEMPLATE_PATH}client-config.js.mustache`,
+    'utf-8'
+  );
   return {
-    hypothesisScript: renderConfig(config.clientUrl),
+    clientUrl: config.clientUrl,
+    hypothesisConfig,
```

Comment:
My understanding is that the top-level `clientUrl` and `hypothesisConfig` properties are only used by `pdfjs-viewer.mustache` and all HTML pages will just use `hypothesisScript`? If so it might be a good idea to move those properties into the PDF-specific serving code to avoid confusion.